cmd/gaia/app/genesis.go

```diff
@@ -155,20 +155,42 @@ func NewDefaultGenesisState() GenesisState {
 // TODO: No validators are both bonded and jailed (#2088)
 // TODO: Error if there is a duplicate validator (#1708)
 // TODO: Ensure all state machine parameters are in genesis (#1704)
-func GaiaValidateGenesisState(genesisState GenesisState) (err error) {
-	err = validateGenesisStateAccounts(genesisState.Accounts)
+func GaiaValidateGenesisState(genesisState GenesisState) error {
+	err := validateGenesisStateAccounts(genesisState.Accounts)
 	if err != nil {
-		return
+		return err
 	}
 	// skip stakeData validation as genesis is created from txs
 	if len(genesisState.GenTxs) > 0 {
 		return nil
 	}
-	return stake.ValidateGenesis(genesisState.StakeData)
+
+	err = stake.ValidateGenesis(genesisState.StakeData)
```

Comment:
A use case for a monad if I ever saw one. Can't wait for Go 2!

Reply:
++